api/v1alpha1/addonimageset_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	ocmv1 "github.com/mt-sre/addon-metadata-operator/pkg/ocm/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AddonImageSetSpec defines the desired state of AddonImageSet
+type AddonImageSetSpec struct {
+	// +kubebuilder:validation:Required
+	// The name of the imageset along with the version.
+	Name string `json:"name" validate:"required"`
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^quay\.io/osd-addons/[a-z-]+`
+	// The url for the index image
+	IndexImage string `json:"indexImage" validate:"required"`
+
+	// +kubebuilder:validation:Required
+	// A list of image urls of related operators
+	RelatedImages []string `json:"relatedImages" validate:"required"`
+
+	// +optional
+	// OCM representation of an add-on parameter
+	AddOnParameters *ocmv1.AddOnParameterList `json:"addOnParameters"`
+
```

Comment:
TODO:
Once [this PR](https://github.com/mt-sre/managed-tenants-cli/pull/42) gets merged, add a `SubscriptionConfig` attribute as well under `AddonImageSpec struct`